eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -1964,6 +1955,11 @@ public function deleteTranslation(ContentInfo $contentInfo, $languageCode)
 
         $this->repository->beginTransaction();
         try {
+            // delete all Versions which contain only single language to be removed
```

Comment:
I think we should iterate over versions here. `singleLangVersions` was initialize outside of the remove operation since it was an error check. 

Now, we must iterate over each version. If it has the language being removed, we test if it has other languages. If it does, we remove the translation from the version. If it only has this language, we remove the version.

Reply:
Hmm, or I could wrap the previously loaded Versions in this try-catch block within Repository transaction to avoid loading again. WDYT @bdunogier ?

*This leads to catching every "expected" API exception and rethrowing after rollback, because PHP doesn't support multi-catch, but since I'm already doing it after this try-catch, I guess it should be ok.*

Reply:
@bdunogier so what I've suggested above is implemented in fixup 3ebea16. Basically wrapped entire loading section into Repository transaction. I'm trying to avoid loading multiple times the same thing. Is this ok with you? If not, then easily revertible as it is fixup :)